core/deployment/src/main/java/io/quarkus/deployment/util/ContainerRuntimeUtil.java

```diff
@@ -70,47 +80,58 @@ public static ContainerRuntime detectContainerRuntime(boolean required) {
         return containerRuntime;
     }
 
-    private static ContainerRuntime getContainerRuntimeEnvironment() {
+    private static ContainerRuntime getContainerRuntimeEnvironment(List<ContainerRuntime> orderToCheckRuntimes) {
         // Docker version 19.03.14, build 5eb3275d40
-        String dockerVersionOutput;
-        boolean dockerAvailable;
+
         // Check if Podman is installed
         // podman version 2.1.1
-        String podmanVersionOutput;
-        boolean podmanAvailable;
+        var runtimesToCheck = new ArrayList<>(orderToCheckRuntimes.stream().distinct().toList());
+        runtimesToCheck.retainAll(List.of(ContainerRuntime.DOCKER, ContainerRuntime.PODMAN));
 
         if (CONTAINER_EXECUTABLE != null) {
-            if (CONTAINER_EXECUTABLE.trim().equalsIgnoreCase("docker")) {
-                dockerVersionOutput = getVersionOutputFor(ContainerRuntime.DOCKER);
-                dockerAvailable = dockerVersionOutput.contains("Docker version");
-                if (dockerAvailable) {
-                    return ContainerRuntime.DOCKER;
-                }
-            }
-            if (CONTAINER_EXECUTABLE.trim().equalsIgnoreCase("podman")) {
-                podmanVersionOutput = getVersionOutputFor(ContainerRuntime.PODMAN);
-                podmanAvailable = PODMAN_PATTERN.matcher(podmanVersionOutput).matches();
-                if (podmanAvailable) {
-                    return ContainerRuntime.PODMAN;
-                }
+            var runtime = runtimesToCheck.stream()
+                    .filter(containerRuntime -> CONTAINER_EXECUTABLE.trim()
+                            .equalsIgnoreCase(containerRuntime.getExecutableName()))
+                    .findFirst()
+                    .filter(r -> {
```

Comment:
I am always surprised when people write this kind of code instead of a simple for each + if statement. Furthermore, I don't understand why we need this and the for loop starting at line `114`

Reply:
> I am always surprised when people write this kind of code instead of a simple for each + if statement

To me this code is far more readable and shows its intention better than the code that was there beforehand. If you prefer it the other way I can re-write it.

> Furthermore, I don't understand why we need this and the for loop starting at line `114`

It was already there before. I didn't really change the structure of the code.

https://github.com/quarkusio/quarkus/blob/1a584779b81dc22741291285a04e0a94c1740f3a/core/deployment/src/main/java/io/quarkus/deployment/util/ContainerRuntimeUtil.java#L82-L99

Reply:
The reason I had to even touch this code is because if I have the `container-image-podman` extension present but I have both docker & podman installed on my machine, it was always finding docker first and trying to build with docker.

This change makes it so the caller can control the search order (if they want to - its backwards compatible to other callers that don't care).

Reply:
> It was already there before. I didn't really change the structure of the code.

Okay, I see that